src/exit_status.rs

```diff
@@ -0,0 +1,30 @@
+/// Add a exit status section
+#[derive(Debug, Clone, Default)]
+pub struct ExitStatus {
+  pub(crate) code: Option<i32>,
+  pub(crate) description: Option<&'static str>,
+  pub(crate) use_default_instead: bool,
```

Comment:
I'm confused. If you have a bunch of these and any one of them has this boolean flag set to true, you'll ignore all information and instead print a list of default statuses? I'd do this with an enum like `enum ExitStatuses { DefaultStatuses, List(Vec<ExitStatus>) }` (with ExitStatus containing fields for code and description)

Reply:
I'm confused and I'm pretty sure the problem is on my end—I'll be the first to admit that I'm a newcomer to Rust and that the vast majority of my programming experience is in languages that lack enums/algebraic types.

But after playing around with the code a bit, I'm not seeing a good way for `ExitStatus` struct to tell the `ExitStatuses` enum that it should be one variant or another without passing some sort of sentinel value like the `use_default_instead` bool.  I'm happy to keep looking, but it would be very helpful if you'd point me in the right direction in terms of finding a way to modify the enum without a sentinel value (which I agree would be cleaner).

After rereading your comment, it occurs to me that you didn't explicitly say not to use the bool sentinel value—it's possible you just meant to set the variant of the enum based on that value rather than setting the `code` and `description` fields directly.  If that is what you meant, then I misunderstood and over-complicated things for no reason!